CMakeLists.txt

```diff
@@ -123,6 +123,18 @@ else ()
     message (AUTHOR_WARNING "${ISSUE_REPORT_MSG}")
 endif ()
 
+# MSVC symbol visibility and linker behavior is sufficiently different from all
+# other platforms. Here, we ensure tests/tools built as EXE choose symbol
+# visibility appropriate for linking the library, whether static or dynamic.
+if (MSVC)
```

Comment:
I cannot say with certainty, but my understanding is that this is MSVC-specific, and therefore tightening works in our favor.

My understanding is that MinGW64 would export all symbols; I'm not sure about Borland.

In this regard, MinGW64 correctness would be improved by this move (I have not confirmed this)

The rationale is to tighten the definition as much as possible prior to 1.0, rather than being too gratuitous.

Also of note: the definition for `NN_EXPORT` has been incorrect for a long time for `NN_STATIC_LIB` on MSVC. It would have been broken altogether for users prior to https://github.com/nanomsg/nanomsg/commit/5b34eb6467975ff4c444e2f8e1267bf01aba4b36#diff-2051e1a7c073348e8ba0043ac9b5c77c, and that commit was not even exactly correct (even though, it "worked")